core/test/variable_operations_test.cpp

```diff
@@ -1278,6 +1278,100 @@ TEST(VariableTest,
+      Dims{Dim::X}, Shape{2}, units::Unit(units::m),
+      Values{Eigen::Vector3d{1, 2, 3}, Eigen::Vector3d{4, 5, 6}});
+  auto y_ = geometry::y(pos);
+  const auto expected_y = makeVariable<double>(
+      Dims{Dim::X}, Shape{2}, units::Unit(units::m), Values{2.0, 5.0});
+  EXPECT_EQ(y_, expected_y);
+}
+
+TEST(VariableTest, unzip_z) {
+  const Variable pos = makeVariable<Eigen::Vector3d>(
+      Dims{Dim::X}, Shape{2}, units::Unit(units::m),
+      Values{Eigen::Vector3d{1, 2, 3}, Eigen::Vector3d{4, 5, 6}});
+  auto z_ = geometry::z(pos);
+  const auto expected_z = makeVariable<double>(
+      Dims{Dim::X}, Shape{2}, units::Unit(units::m), Values{3.0, 6.0});
+  EXPECT_EQ(z_, expected_z);
+}
+TEST(VariableTest, zip_unzip_positions) {
+  const Variable x_in = makeVariable<double>(
+      Dims{Dim::X}, Shape{3}, units::Unit(units::m), Values{1, 2, 3});
+  const Variable y_in = makeVariable<double>(
+      Dims{Dim::X}, Shape{3}, units::Unit(units::m), Values{4, 5, 6});
+  const Variable z_in = makeVariable<double>(
+      Dims{Dim::X}, Shape{3}, units::Unit(units::m), Values{7, 8, 9});
+  auto positions = geometry::position(x_in, y_in, z_in);
+  auto x_out = geometry::x(positions);
+  auto y_out = geometry::y(positions);
+  auto z_out = geometry::z(positions);
+  EXPECT_EQ(x_in, x_out);
+  EXPECT_EQ(y_in, y_out);
+  EXPECT_EQ(z_in, z_out);
+}
+
+TEST(VariableTest, rotate) {
+  Eigen::Vector3d vec1(1, 2, 3);
+  Eigen::Vector3d vec2(4, 5, 6);
+  const Variable vec = makeVariable<Eigen::Vector3d>(
+      Dims{Dim::X}, Shape{2}, units::Unit(units::m), Values{vec1, vec2});
+  Eigen::Quaterniond rot1(1.1, 2.2, 3.3, 4.4);
+  Eigen::Quaterniond rot2(5.5, 6.6, 7.7, 8.8);
+  rot1.normalize();
+  rot2.normalize();
+  const Variable rot = makeVariable<Eigen::Quaterniond>(
+      Dims{Dim::X}, Shape{2}, units::Unit(units::dimensionless),
+      Values{rot1, rot2});
+  auto vec_new = geometry::rotate(vec, rot);
+  const auto rotated = makeVariable<Eigen::Vector3d>(
+      Dims{Dim::X}, Shape{2}, units::Unit(units::m),
+      Values{rot1._transformVector(vec1), rot2._transformVector(vec2)});
+  EXPECT_EQ(vec_new, rotated);
+}
+
+TEST(VariableTest, rotate_inplace) {
+  Eigen::Vector3d vec1(1, 2, 3);
+  Eigen::Vector3d vec2(4, 5, 6);
+  auto vec = makeVariable<Eigen::Vector3d>(
+      Dims{Dim::X}, Shape{2}, units::Unit(units::m), Values{vec1, vec2});
+  Eigen::Quaterniond rot1(1.1, 2.2, 3.3, 4.4);
+  Eigen::Quaterniond rot2(5.5, 6.6, 7.7, 8.8);
+  rot1.normalize();
+  rot2.normalize();
+  const Variable rot = makeVariable<Eigen::Quaterniond>(
+      Dims{Dim::X}, Shape{2}, units::Unit(units::dimensionless),
+      Values{rot1, rot2});
+  VariableView vec_new = geometry::rotate(vec, rot, vec);
+  const auto rotated = makeVariable<Eigen::Vector3d>(
+      Dims{Dim::X}, Shape{2}, units::Unit(units::m),
+      Values{rot1._transformVector(vec1), rot2._transformVector(vec2)});
+  EXPECT_EQ(vec_new, rotated);
+  EXPECT_EQ(vec, rotated);
```

Comment:
Merge inplace code into same test?
```suggestion
  VariableView vec_new2 = geometry::rotate(vec, rot, vec);
  EXPECT_EQ(vec_new2, rotated);
  EXPECT_EQ(vec, rotated);
```